controllers/configuration_controller_test.go

```diff
@@ -1569,6 +1572,16 @@ func TestTerraformDestroy(t *testing.T) {
 			objects:          []client.Object{readyProvider, configurationWithConnSecret, baseConfigurationCM, completeDestroyJob, baseVariableSecret, connectionSecret},
 			deletedResources: []client.Object{baseConfigurationCM, completeDestroyJob, baseVariableSecret, connectionSecret},
 		},
+		{
+			name: "destroy job has completes, cleanup resources but backend secret",
+			args: args{
+				configuration: configurationWithConnSecret,
+				meta:          &metaWithDeleteResourceIsFalse,
+			},
+			want:             want{},
+			objects:          []client.Object{readyProvider, configurationWithConnSecret, baseConfigurationCM, completeDestroyJob, baseVariableSecret, connectionSecret},
+			deletedResources: []client.Object{baseConfigurationCM, completeDestroyJob, baseVariableSecret, connectionSecret},
```

Comment:
Backend secret should be added to `keptResources` to test this PR. I think now this condition is not tested.

Reply:
I missed this test case, and I will add it.